File: lib/java/ReportFormatter.java
package lib.java;

import java.time.OffsetDateTime;
import java.time.format.DateTimeFormatter;

public class ReportFormatter {
  public static String serializeOrder(String orderId, double amount) {
    return orderId + ":" + String.format("%.2f", amount);
  }

  public static String formatDate(String value) {
    return OffsetDateTime.parse(value).format(DateTimeFormatter.ISO_LOCAL_DATE);
  }

  // Non-public: should not be treated as exported/public symbol.
  static String formatDateDebug(String value) {
    return value;
  }
}

File: lib/go/report_formatter.go
package reportformatter

import (
	"fmt"
	"time"
)

// SerializeOrder is intentionally duplicated by name across languages.
func SerializeOrder(orderID string, amount float64) string {
	return fmt.Sprintf("%s:%.2f", orderID, amount)
}

func FormatDate(raw string) string {
	parsed, err := time.Parse(time.RFC3339, raw)
	if err != nil {
		return raw
	}
	return parsed.Format("2006-01-02")
}

func renderHtmlCard(userHTML string) string {
	return "<div>" + userHTML + "</div>"
}
